lib/create-contentful-api.ts

```diff
@@ -656,30 +770,44 @@ export default function createContentfulApi<OptionType>(
     if (isClientWithAllLocalesAndWithoutLinkResolution(options)) {
       return getEntriesWithAllLocalesAndWithoutLinkResolution<Fields>(query)
     }
-    if (isClientWithAllLocalesAndWithLinkResolution(options)) {
-      return getEntriesWithAllLocalesAndWithLinkResolution<Fields>(query)
+    if (isClientWithAllLocalesAndWithLinkResolutionAndWithUnresolvableLinks(options)) {
+      return getEntriesWithAllLocalesAndWithLinkResolutionAndWithUnresolvableLinks<Fields>(query)
+    }
+    if (isClientWithAllLocalesAndWithLinkResolutionAndWithoutUnresolvableLinks(options)) {
+      return getEntriesWithAllLocalesAndWithLinkResolutionAndWithoutUnresolvableLinks<Fields>(query)
     }
     if (isClientWithoutLinkResolution(options)) {
       return getEntriesWithoutLinkResolution<Fields>(query)
     }
-    if (isClientWithLinkResolution(options)) {
-      return getEntriesWithLinkResolution<Fields>(query)
+    if (isClientWithLinkResolutionAndWithUnresolvableLinks(options)) {
+      return getEntriesWithLinkResolutionAndWithUnresolvableLinks<Fields>(query)
+    }
+    if (isClientWithLinkResolutionAndWithoutUnresolvableLinks(options)) {
+      return getEntriesWithLinkResolutionAndWithoutUnresolvableLinks<Fields>(query)
     }
     return getEntriesDefault<Fields>(query)
   }
 
   async function internalGetEntries<RType>(
     query: Record<string, any>,
-    resolveLinks = false
+    options: {
+      withoutLinkResolution: boolean
+      withoutUnresolvableLinks?: boolean
+    }
   ): Promise<RType> {
-    const { removeUnresolved } = getGlobalOptions()
+    const { withoutLinkResolution, withoutUnresolvableLinks } = options
+
     try {
       const entries = await get({
         context: 'environment',
         path: 'entries',
         config: createRequestConfig({ query: normalizeSelect(query) }),
       })
-      return resolveCircular(entries, { resolveLinks, removeUnresolved }) as RType
+
+      return resolveCircular(entries, {
+        resolveLinks: !withoutLinkResolution,
+        removeUnresolved: !!withoutUnresolvableLinks,
```

Comment:
I would be more specific with the default value, then we also don't need the double negation: 

```suggestion
        resolveLinks: !withoutLinkResolution ?? true,
        removeUnresolved: withoutUnresolvableLinks ?? false,
```

Reply:
I believe that `resolveLinks` will always have a value, and the `!!` it's just casting to boolean which cover the `undefined` case, I feel that adding default value it's a bit redundant 🤔

Reply:
True, but I think if one of them can be `undefined`, it makes sense to have a default value, and to be consistent I would also give a default value to the other one (I agree it's not really necessary for that one). 
Plus, it implicitly documents the default behavior of the contentful.js library (i.e. returning entries with link resolution and with unresolvable links)